rust-toolchain.toml

```diff
@@ -1,9 +1,5 @@
 [toolchain]
-# Pinned due to https://github.com/rust-osdev/uefi-rs/issues/500.
```

Comment:
hm, I'm not sure if we want to have this.

pro: We will be notified about upstream breakage early
cons: Might brings us build trouble in the future again.

Personally, I'd tend to an "update the nightly version to a specific one"-approach on a regular base. What do you think?

Reply:
The major advantage to using an unspecified nightly is that it allows us to act as the CI for Rust's UEFI targets. The Rust project itself doesn't run any CI for those targets currently. I have a [PR](https://github.com/rust-lang/rust/pull/101703) up to add some minimal CI for x86_64-unknown-uefi, but that is blocked until the UEFI targets become tier 2.

So at least until that situation changes, I would like us to use currently nightly so that we can be notified when upstream breaks the UEFI targets so that we can go fix the upstream problem.

Once the upstream targets do get CI, I think it would still be good for us to test nightly for as long as we rely on unstable features, because a nightly change could break those unstable features at any time (sometimes due to a bug, but sometimes due to an intentional change in the unstable feature, requiring us to make code changes).

Now that we have the toolchain file it's not too much trouble to deal with these nightly breakages anyway; we'll just pop up a PR to modify the toolchain file to the previous day's nightly and file an issue.

Reply:
Alright. I agree :)